### PR TITLE
[MIG] partner_contact_lang: Migration to 11.0

### PR DESCRIPTION
**Pull request by cristinamartinrod**
Odoo by default propagate language field to the created contacts from their form, but it doesn't allow to change it once created. This module fills this gap, and also provides other facilities for the
contact language management:

* Put the language of the parent company when the contact doesn't have a language and this parent company is assigned.
* When the company changes the language, it fills with the same language all the contacts that don't have any.

cc @Tecnativa

### CODE REVIEW

File: partner_contact_lang/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+{
+    'name': 'Manage language in contacts',
+    'version': '11.0.1.0.0',
+    'category': 'Sales',
+    'license': 'AGPL-3',
+    'author': 'Tecnativa,'
+              'Odoo Community Association (OCA)',
+    'website': 'https://github.com/OCA/partner-contact/',
+    'depends': [
+        'base',
```
Comment (nikul-serpentcs) [inline partner_contact_lang/__manifest__.py:14]: @cristinamartinrod `base` is a default depends, It's Remove it.

File: partner_contact_lang/tests/test_partner_contact_lang.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    def test_onchange_parent_id(self):
+        self.contact.parent_id = False
+        res = self.contact.onchange_parent_id()
+        self.assertIsNone(res)
+        self.contact.parent_id = self.partner
+        res = self.contact.onchange_parent_id()
+        self.assertEqual(res.get('value', {}).get('lang'), 'en_US')
+
+    def test_write_parent_lang(self):
+        self.partner.lang = False
```
Comment (nikul-serpentcs) [inline partner_contact_lang/tests/test_partner_contact_lang.py:36]: @cristinamartinrod Why `self.partner.lang` false here? 
And Added [here]( https://github.com/OCA/partner-contact/pull/636/files#diff-81fadbe02bdd60101996d81bd5742cbaR33), In created time You have added already [here](https://github.com/OCA/partner-contact/pull/636/files#diff-81fadbe02bdd60101996d81bd5742cbaR15)

File: partner_contact_lang/tests/test_partner_contact_lang.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        self.partner.lang = 'en_US'
+        self.assertEqual(self.contact.lang, 'en_US')
```
Comment (nikul-serpentcs) [inline partner_contact_lang/tests/test_partner_contact_lang.py:38]: replace `contact` to `partner`

**Review by nikul-serpentcs — COMMENTED** [review]
Improve Code

File: partner_contact_lang/__manifest__.py (hunk 1 above)
Comment (pedrobaeza) [reply partner_contact_lang/__manifest__.py:14]: You always need to depend at least in this one. If not, bad things can happen.

File: partner_contact_lang/tests/test_partner_contact_lang.py (hunk 2 above)
Comment (pedrobaeza) [reply partner_contact_lang/tests/test_partner_contact_lang.py:36]: It's for using same setup record for testing other things

File: partner_contact_lang/tests/test_partner_contact_lang.py (hunk 3 above)
Comment (pedrobaeza) [reply partner_contact_lang/tests/test_partner_contact_lang.py:38]: We want to test the propagation of the parent value, so it's correct

File: partner_contact_lang/tests/test_partner_contact_lang.py (hunk 2 above)
Comment (nikul-serpentcs) [reply partner_contact_lang/tests/test_partner_contact_lang.py:36]: Yes, But here already added in creating time, If  Created Partner then already add lang.

File: partner_contact_lang/tests/test_partner_contact_lang.py (hunk 3 above)
Comment (nikul-serpentcs) [reply partner_contact_lang/tests/test_partner_contact_lang.py:38]: IMO Ok then here check both(contact to partner).

Comment (pedrobaeza) [reply partner_contact_lang/tests/test_partner_contact_lang.py:38]: OK, not specially needed, but OK

File: partner_contact_lang/tests/test_partner_contact_lang.py (hunk 2 above)
Comment (pedrobaeza) [reply partner_contact_lang/tests/test_partner_contact_lang.py:36]: This is for being sure this has been emptied before, but OK to remove it.

**Comment by pedrobaeza** [comment]
@cristinamartinrod please make the agreed changes

**Review by nikul-serpentcs — APPROVED** [review]
Code Review LGTM

**Comment by christophe-hanon** [comment]
Just did a small functionnal review

Readme describes correctly the module
Language is propagated to children - only if the language is not set.
Checked with two languages.
Please merge.

**Review by christophe-hanon — COMMENTED** [review]
Just did a small functionnal review

Readme describes correctly the module
Language is propagated to children - only if the language is not set.
Checked with two languages.
Please merge.

File: partner_contact_lang/tests/test_partner_contact_lang.py [hunk 4]
```diff
@@ -29,6 +29,5 @@ def test_onchange_parent_id(self):
         self.assertEqual(res.get('value', {}).get('lang'), 'en_US')
 
     def test_write_parent_lang(self):
-        self.partner.lang = False
         self.partner.lang = 'en_US'
-        self.assertEqual(self.contact.lang, 'en_US')
+        self.assertEqual(self.partner.lang, 'en_US')
```
Comment (tafaRU) [inline partner_contact_lang/tests/test_partner_contact_lang.py]: why did you change the unit test?

Comment (nikul-serpentcs) [reply partner_contact_lang/tests/test_partner_contact_lang.py]: @tafaRU Because [here](https://github.com/OCA/partner-contact/pull/636#discussion_r222618424) two times checked `contact`, so changed `partner`. 

assertion checked one for `partner` and one for `contact`.

Comment (pedrobaeza) [reply partner_contact_lang/tests/test_partner_contact_lang.py]: This one is incorrect right now and it's not testing what the module does. I'll change it.

**Review by tafaRU — APPROVED** [review]
